core/src/main/java/org/openstack4j/openstack/barbican/domain/BarbicanSecret.java

```diff
@@ -0,0 +1,168 @@
+package org.openstack4j.openstack.barbican.domain;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.MoreObjects;
+import org.openstack4j.model.barbican.Secret;
+import org.openstack4j.model.barbican.builder.SecretCreateBuilder;
+import org.openstack4j.openstack.common.ListResult;
+
+import java.util.Date;
+import java.util.List;
+
+/**
+ * Created by reneschollmeyer on 02.08.17.
+ */
+public class BarbicanSecret implements Secret {
+
+    private String name;
+    @JsonProperty("algorithm")
```

Comment:
Nit: `@JsonProperty("<jsonName>")` is optional, if property name is equal in json and java